src/CosmosDB/CosmosDB/Restore/GetAzCosmosDBSqlRestorableResource.cs

```diff
@@ -0,0 +1,76 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.CosmosDB.Helpers;
+using Microsoft.Azure.Commands.CosmosDB.Models;
+using Microsoft.Azure.Management.CosmosDB.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Newtonsoft.Json.Serialization;
+using System;
+using System.Collections;
+using System.Management.Automation;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Commands.CosmosDB
+{
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "CosmosDBSqlRestorableResource", DefaultParameterSetName = NameParameterSet), OutputType(typeof(PSDatabaseToRestore))]
+    public class GetAzCosmosDBSqlRestorableResource : AzureCosmosDBCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = Constants.LocationNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string LocationName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.AccountInstanceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string DatabaseAccountInstanceId { get; set; }
+
+        [Newtonsoft.Json.JsonConverter(typeof(IsoDateTimeConverter))]
+        [Parameter(Mandatory = true, HelpMessage = Constants.RestoreTimestampHelpMessage)]
+        public DateTime RestoreTimestampInUtc { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = Constants.RestoreLocationNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string RestoreLocation { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = ParentObjectParameterSet, HelpMessage = Constants.RestorableDatabaseAccountObjectHelpMessage)]
+        [ValidateNotNull]
+        public PSRestorableDatabaseAccountGetResult ParentObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ParameterSetName.Equals(ParentObjectParameterSet, StringComparison.Ordinal))
+            {
+                LocationName = ParentObject.Location;
+                DatabaseAccountInstanceId = ParentObject.DatabaseAccountInstanceId;
+            }
+
+            DateTime dateTimeInUtc;
+            if (RestoreTimestampInUtc.Kind == DateTimeKind.Unspecified)
+            {
+                dateTimeInUtc = RestoreTimestampInUtc;
+            }
+            else
+            {
+                dateTimeInUtc = RestoreTimestampInUtc.ToUniversalTime();
+            }
+
+            WriteObject("RestoreTimestamp: " + RestoreTimestampInUtc);
+            WriteObject("RestoreTimestamp in UTC: " + RestoreTimestampInUtc.ToUniversalTime());
```

Comment:
Not sure of how good an idea it is to Write strings on console. (Your output type should also contain this information). IMO, you should remove this and add an elaborate help message explaining that if DateTimeKind is unspecified then it will converted to UTC

Reply:
My bad. these were for debugging. Fixed all of them now.